picard/util/pipe.py

```diff
@@ -94,14 +98,17 @@ def __init__(self, app_name: str, app_version: str, args=None):
                         break
 
     def __generate_filename(self) -> str:
-        prefix: str = ""
-        username: str = os.getlogin()
         if self.__is_win:
             prefix = "\\\\.\\pipe\\"
+        elif self.__is_mac:
+            prefix = os.path.expanduser("~/Library/Application Support/MusicBrainz/Picard/pipes/")
         else:
-            prefix = "/tmp/"
+            prefix = f"{os.getenv('XDG_RUNTIME_DIR')}/"
+            # just in case the $XDG_RUNTIME_DIR is not declared, fallback dir
+            if not prefix:
+                prefix = os.path.expanduser("~/.config/MusicBrainz/Picard/pipes/")
```

Comment:
I don't think parent directories aren't created by `os.mkfifo()`, and those directories may not exist, so this needs to be handled.

Reply:
You're right. Non-existent paths raise `FileNotFoundError`

Reply:
Should work fine after latest commits.